frontend/src/SinglePage.js

```diff
@@ -19,19 +20,51 @@ const averageRatings = (ratings) => {
 class SinglePage extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      data: [],
+      user: [],
+      saved: false,
+      checkedSave: false
+    };
   }
 
-  componentDidMount() {
-    fetch(`${backendHostURL}/hike/${this.props.match.params.id}`)
+  async componentDidMount() {
+    await fetch(`${backendHostURL}/hike/${this.props.match.params.id}`)
       .then((res) => res.json())
       .then((data) => {
         console.log("data", data);
-        this.setState({ ...data[0] });
+        this.setState({ data: data[0] });
         document.getElementById("rating-num").innerText = averageRatings(
-          this.state.rating
+          this.state.data.rating
         );
       });
+    this.checkSaved()
+    this.state.checkedSave = true
+    console.log("checked save")
+  }
+
+  async checkSaved() {
+    await fetch(`${backendHostURL}/login/${localStorage.getItem("_id")}`)
+      .then((res) => res.json())
+      .then((data) => {
+        this.setState({ user: this.state.user.concat(data[0]) });
+        console.log("user", this.state.user)
+        if(this.state.user.length > 0)
+        {          
+          const hike = this.state.data._id
+          const length = this.state.user[0].saved_trails.length
+          for(let i = 0; i < length; i++)
+          {
+              const curr = this.state.user[0].saved_trails[i]
+              console.log(curr)
```

Comment:
maybe we should remove some console logs 😅

Reply:
True! I thought I deleted most of them but there were definitely some still left behind